libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1427,13 +1429,36 @@ void SMTEncoder::bitwiseOperation(BinaryOperation const& _op)
 	optional<smtutil::Expression> result;
 	if (_op.getOperator() == Token::BitAnd)
 		result = bvLeft & bvRight;
-	// TODO implement the other operators
-	else
-		m_errorReporter.warning(
-			1093_error,
-			_op.location(),
-			"Assertion checker does not yet implement this bitwise operator."
-		);
+	else if (_op.getOperator() == Token::BitOr)
+		result = bvLeft | bvRight;
+	else if (_op.getOperator() == Token::BitXor)
+		result = bvLeft ^ bvRight;
+
+	if (result)
+		defineExpr(_op, smtutil::Expression::bv2int(*result, isSigned));
+}
+
+void SMTEncoder::bitwiseNotOperation(UnaryOperation const& _op)
+{
+	solAssert(_op.getOperator() == Token::BitNot, "");
+
+	unsigned bvSize = 256;
+	bool isSigned = false;
+	if (auto const* intType = dynamic_cast<IntegerType const*>(_op.annotation().type))
+	{
+		bvSize = intType->numBits();
+		isSigned = intType->isSigned();
+	}
+	else if (auto const* fixedType = dynamic_cast<FixedPointType const*>(_op.annotation().type))
+	{
+		bvSize = fixedType->numBits();
+		isSigned = fixedType->isSigned();
+	}
+	else if (auto const* fixedBytesType = dynamic_cast<FixedBytesType const*>(_op.annotation().type))
+		bvSize = fixedBytesType->numBytes() * 8;
```

Comment:
This block that computes `bvSize` and `isSigned` is repeated from `bitwiseOperation`. Maybe the two functions could be merged and adjusted accordingly

Reply:
I can merge them, in that case parameter would be `Expression` (one has `BinaryOperator` argument, other has `UnaryOperator`). Other option is to create helper function that calculates `bvSize` and `isSigned` and called from these two.

Reply:
Implemented helper function, hope it is more acceptable now, as there is no code duplication regarding `bvSize` and `isSigned`. Let me know if it is better other way.